### ISSUE TITLE
Octotiger does not scale to 2 knl nodes on Cori

### ISSUE THREAD

**Issue by dmarce1** [STEllAR-GROUP/octotiger#38]
When running Octotiger on Cori knl,quad,cache, the moving_star problem with -Xscale 32.0 -max_level=5 takes about 4 seconds per timestep on one node, but about 40 seconds per timestep on 2 nodes.

This is using the commands: 
srun -n 2 -c 68 -u ./octotiger -t 68 -Problem=moving_star -Xscale=32.0 -Max_level=5 -Disableoutput
and
srun -n 1 -c 68 -u ./octotiger -t 68 -Problem=moving_star -Xscale=32.0 -Max_level=5 -Disableoutput

This is with with tcmalloc.

**Comment by dmarce1** [comment]
Here is a link to vtune profiles for the two node run. 
https://drive.google.com/file/d/0B_Hf1bEwvJEkeXJicGRKQ2RxR3M/view?usp=sharing

**Comment by DavidPfander-UniStuttgart** [comment]
I think at this point this is to be expected, as we already scale badly on a single node. 

(I made similar observations when I ran only multiple nodes two weeks ago.)

**Comment by biddisco** [comment]
Last night I ran number of tests of the verbs PP on the greina machine at CSCS where I have been doing the majority of my development.

I’m seeing performance drops of my 2 node latency tests of quite significant magnitude relative to the results I collected 4 months ago – unfortunately, I do not have any saved timings from more recently and I have been blindly running tests (to fix races etc) without comparing absolute numbers for the most part.

In sept/oct I was getting latencies of 1.5us, last night, the best I could manage was around 10us

When I increase the number of threads, the performance drops significantly, which indicates to me that there is some false sharing or other thread contention that is causing major performance problems – particularly around the parcelport area.

I can’t yet offer any further insights other than I suspect some catastrophic failure on my part to get stuff working properly on my machines.

JB

**Comment by biddisco** [comment]
PS. I forgot to add that I went back (git date) to about jan 10th with my tests, but not further than that.

JB

From: cscs <biddisco@cscs.ch<mailto:biddisco@cscs.ch>>
Date: Thursday 26 January 2017 at 12:58
To: STEllAR-GROUP/octotiger <reply@reply.github.com<mailto:reply@reply.github.com>>
Subject: Re: [STEllAR-GROUP/octotiger] Octotiger does not scale to 2 knl nodes on Cori (#38)

Last night I ran number of tests of the verbs PP on the greina machine at CSCS where I have been doing the majority of my development.

I’m seeing performance drops of my 2 node latency tests of quite significant magnitude relative to the results I collected 4 months ago – unfortunately, I do not have any saved timings from more recently and I have been blindly running tests (to fix races etc) without comparing absolute numbers for the most part.

In sept/oct I was getting latencies of 1.5us, last night, the best I could manage was around 10us

When I increase the number of threads, the performance drops significantly, which indicates to me that there is some false sharing or other thread contention that is causing major performance problems – particularly around the parcelport area.

I can’t yet offer any further insights other than I suspect some catastrophic failure on my part to get stuff working properly on my machines.

JB
